programs/marginfi/src/instructions/marginfi_account/repay.rs

```diff
@@ -119,8 +133,7 @@ pub struct LendingAccountRepay<'info> {
 
     #[account(
         mut,
-        has_one = group,
-        has_one = authority
```

Comment:
Surely it's the opposite?
- `authority` shoule be renamed to `signer` because it's no longer the marginfi account authority necessarily. 
- Add the account authority constraint check above the account, something like `mfiaccount.load().is_in_receivership || mfiaccount.load().authority == signer.key()` 

And no, of course we should not let this be a permissionless instruction. We should be defaulting to limited use instructions to avoid unintended side effects.